src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/service/AsyncActionPlanEventService.kt

```diff
@@ -16,14 +16,22 @@ private val log = KotlinLogging.logger {}
  */
 @Component
 class AsyncActionPlanEventService(
+  private val telemetryService: TelemetryService,
   private val timelineEventFactory: TimelineEventFactory,
   private val timelineService: TimelineService,
 ) : ActionPlanEventService {
   override fun actionPlanCreated(actionPlan: ActionPlan) {
     runBlocking {
       log.info { "ActionPlan created event for prisoner [${actionPlan.prisonNumber}]" }
-      val timelineEvent = timelineEventFactory.actionPlanCreatedEvent(actionPlan)
-      launch { timelineService.recordTimelineEvent(actionPlan.prisonNumber, timelineEvent) }
+      launch {
```

Comment:
Minor, and it probably doesnt add much (if anything), but I wonder if the launch could/should be in the loop?
```
      actionPlan.goals.forEach {
        launch {
          telemetryService.trackGoalCreateEvent(it)
        }
      }
```

There's probably nothing in it, and I dont think either approach would affect the time take for the user (ie. either do track events in a thread, or track each event in its own thread)

Reply:
Ignore me - its probably a premature optimisation